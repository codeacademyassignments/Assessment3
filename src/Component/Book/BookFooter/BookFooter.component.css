
.heart{
  width:100%;
}

.heartImage{
  align-self: center;
  width:10%;
}
.cardFooter{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0% 4%;

}
