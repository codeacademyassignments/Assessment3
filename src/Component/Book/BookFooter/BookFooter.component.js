import React from 'react';
import './BookFooter.component.css';
import PropTypes from 'prop-types';

const CardFooter = (props) => {
  const heartUrl = props.liked ? 'heart-red.svg' : 'heart-white.svg';
  return (
    <div className="cardFooter">
      <div>
        <p className="rating">{props.rating}</p>
      </div>
      <div className="heartImage">
        <img
          src={require(`../../../Icons/${heartUrl}`)}
          alt="heart-black"
          className="heart"
          onClick={() => {
            props.onHeartClick(props.liked);
          }}
        />
      </div>
    </div>
  );
};

CardFooter.propTypes = {
  rating: PropTypes.number.isRequired,
  liked: PropTypes.bool.isRequired,
};

export default CardFooter;
