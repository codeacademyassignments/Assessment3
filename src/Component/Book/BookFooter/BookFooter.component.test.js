import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import BookFooter from './BookFooter.component';


describe('BookFooter', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<BookFooter liked title="Harry Potter and prisoner of azekaban" rating={4.3} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('mock function', () => {
  const onHeartClickMock = jest.fn();
  const wrapper = shallow(<BookFooter liked rating={4.5} onHeartClick={onHeartClickMock} />);
  it('should call onHeartClickMock when heart clicked', () => {
    wrapper.find('.heart').simulate('click');
    expect(onHeartClickMock).toHaveBeenCalledWith(true);
    expect(onHeartClickMock).toHaveBeenCalledTimes(1);
  });
});
