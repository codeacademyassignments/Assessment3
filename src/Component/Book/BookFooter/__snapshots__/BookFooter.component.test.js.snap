// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookFooter snapshot should match 1`] = `
<div
  className="cardFooter"
>
  <div>
    <p
      className="rating"
    >
      4.3
    </p>
  </div>
  <div
    className="heartImage"
  >
    <img
      alt="heart-black"
      className="heart"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "heart-red.svg",
        }
      }
    />
  </div>
</div>
`;
